src/main/java/randoop/JunitFileWriter.java

```diff
@@ -351,46 +352,61 @@ public File writeDriverFile() {
-
       NameGenerator instanceNameGen = new NameGenerator("t");
+      Map<String, String> testClassToTestVariable = new HashMap<>();
       for (String testClass : testClassNames) {
-
         if (beforeAllText != null) {
           out.println();
-          out.println("    " + testClass + "." + BEFORE_ALL_METHOD + "();");
+          out.println("  " + testClass + "." + BEFORE_ALL_METHOD + "();");
           out.println();
         }
-
         String testVariable = instanceNameGen.next();
-        out.println("    " + testClass + " " + testVariable + "= new " + testClass + "();");
+        testClassToTestVariable.put(testClass, testVariable);
+        out.println("  static " + testClass + " " + testVariable + "= new " + testClass + "();");
+      }
+      out.println("  static boolean wasSuccessful = true;");
+      out.println("  public static void main(String[] args) {");
+
+      int totalCallCount = 0;
+      NameGenerator inClassMethodGen = new NameGenerator("method", 1, numDigits(10));
+      for (String testClass : testClassNames) {
 
         int classMethodCount = classMethodCounts.get(testClass);
         NameGenerator methodGen = new NameGenerator("test", 1, numDigits(classMethodCount));
 
         while (methodGen.nameCount() < classMethodCount) {
           out.println();
+          // TODO discuss what before each text represents.
           if (beforeEachText != null) {
-            out.println("    " + testVariable + "." + BEFORE_EACH_METHOD + "();");
+            out.println(
+                "    " + testClassToTestVariable.get(testClass) + "." + BEFORE_EACH_METHOD + "();");
           }
           String methodName = methodGen.next();
           out.println("    try {");
-          out.println("      " + testVariable + "." + methodName + "();");
+          out.println("      " + testClassToTestVariable.get(testClass) + "." + methodName + "();");
           out.println("    } catch (Throwable e) {");
           out.println("      wasSuccessful = false;");
           out.println("      e.printStackTrace();");
           out.println("    }");
           if (afterEachText != null) {
-            out.println("    " + testVariable + "." + AFTER_EACH_METHOD + "();");
+            out.println(
+                "    " + testClassToTestVariable.get(testClass) + "." + AFTER_EACH_METHOD + "();");
+          }
+          totalCallCount += 1;
+          if (totalCallCount == 500) {
+            // means we need to create a new method and call that
+            String classMethodName = inClassMethodGen.next();
+            out.println("    " + classMethodName + "();");
+            out.println("  }");
+            out.println();
+            out.println("  public static void " + classMethodName + "() {");
+            totalCallCount = 0;
           }
         }
 
         if (afterAllText != null) {
           out.println("    " + testClass + "." + AFTER_ALL_METHOD + "();");
         }
-
```

Comment:
Could you please remove unnecessary changes from your fork?
That will make the diffs smaller to read and, once merged, will make the version control history smaller and cleaner.
Thanks!